lib/ai_computer_player.py

```diff
@@ -0,0 +1,74 @@
+from .tictactoe_board import *
+import time
+import math
+
+
+class AiComputerPlayer:
+    def __init__(self, board, mark, ui):
+        self.board = board
+        self.mark = mark
+        self.ui = ui
+
+    def move(self):
+        self.ui.display("The computer is thinking.....")
+        time.sleep(1)
+        index = self.ai_index()
+        self.board.mark_position(index, self.mark)
+        self.board.display_board()
+
+    def is_a_winner(self, board):
+        for win_comb in self.board.win_combinations:
+            if (
+                board[win_comb[0]] == board[win_comb[1]] == board[win_comb[2]]
+                and board[win_comb[0]] != " "
+            ):
+                return True
+            else:
+                return False
+
+    def get_winner(self, board, player):
+        for win_comb in self.board.win_combinations:
+            if board[win_comb[0]] == board[win_comb[1]] == board[win_comb[2]] == player:
+                return player
+
+    def is_board_full(self, board):
```

Comment:
Might have missed this one, but use the `is_board_full` method coming from the `Board` class. One issue that might arise is if the `EMPTY_SPACE` value were to change from `" "` to `nil`, then this method would no longer work.

Reply:
Here I see a problem...if I will use `is_board_full` from  `Board` class I will break `minimax` method  because `is_board_full` is checking the board instance. I think the best fix will be to make `is_board_full` method from `Board` class more abstract by taking a general board(dictionary) as a  parameter and than I can use it in `AiComputerPlayer` class too